common.blocks/version-selector/version-selector.js

```diff
@@ -20,15 +20,15 @@ modules.define('version-selector', [ 'i-bem__dom', 'querystring' ], function(pro
                         var ver = encodeURIComponent(qs.parse(location.href).version || '');
 
                         if (!ver) {
-                            ver = JSON.parse(select.dataset.bem);
-                            ver = ver['version-selector'].default.hash;
+                            ver = select.params.default.name;
                         }
 
                         select.setValue(ver);
                     };
 
                     select.domElem.on('change', function(e) {
-                        var val = e.target.value;
+                        var name = e.target.value;
+                        var hash = select._getHashByVersion(name);
```

Comment:
BTW, since `e.target.value` used only in getHashByVersion, we can write much more simple code:

```js
var getHashByVersion = function(item) { return item.name === e.target.value; };
var hash = this.params.versions.find(getHashByVersion) || null;
```